### ISSUE TITLE
Selecting fields not working for nested relationship.

### ISSUE THREAD

**Issue by hemratna** [spatie/laravel-query-builder#317]
Let's consider below use case.

`User` has O2M `Campaigns`, and `Campaigns` has O2M `RedirectionUrls `.

```
QueryBuilder::for(User::class)
            ->allowedIncludes('campaigns,campaigns.redirectionurls')
            ->allowedFields('name', 'uuid', 'id', 'campaigns.product_url', 'campaigns.created_by', 'campaigns.id', 'campaigns.redirectionurls.id', 'campaigns.redirectionurls.campaign_id')
            ->paginate();
```

When I tried the below GET request,
`/users/4?include=campaigns,campaigns.redirectionurls&fields[users]=name,uuid,id,created_at&fields[campaigns]=product_url,created_by,id&fields[campaigns.redirectionurls]=url` 

the response will show all the fields for the `redirection_urls` instead of showing the `URL` only. 

In other words, the `fields` will work fine for `users`, `campaigns` But didn't work for `campaigns.redirectionurls`

Let me know if I am making any mistake to pass the arguments.

**Comment by ChiselspeedTuning** [comment]
I think, but could be wrong you may need to create a custom filter to filter through a nested relationship.

**Comment by hemratna** [comment]
@ChiselspeedTuning It will really help if you give me some reference example. 
Thanks for looking into this.

**Comment by ChiselspeedTuning** [comment]
@hemratna My apologies, I misunderstood on what you was wanting to achieve, a custom filter is the wrong way to go. I do have an idea on how to achieve what you are wanting to achieve, but I will need to try it out first. I will get back to you on this if nobody else does before I get the chance to.

**Comment by hemratna** [comment]
@chapeupreto Thanks a lot for looking into this. Waiting for your response.

**Comment by AlexVanderbist** [comment]
Hi @hemratna, this might be an issue on our end. Usually when including relationships we look at the allowedfields to determine what fields to load with the included relationship. This is tested to work for at least 1 "level" deep relationships. I'll try to add a test for including fields on includes that are > 1 levels deep today and let you know what the results are.

Btw, just to be sure: can you try moving `allowedFields` before `allowedIncludes`?

**Comment by hemratna** [comment]
@AlexVanderbist I tried with 
```
QueryBuilder::for(User::class)
            ->allowedFields('name', 'uuid', 'id', 'campaigns.product_url', 'campaigns.created_by', 'campaigns.id', 'campaigns.redirectionurls.id', 'campaigns.redirectionurls.campaign_id')
			->allowedIncludes('campaigns,campaigns.redirectionurls')
            ->paginate();
```

But still not working.
Thanks a lot for looking into this.

**Comment by AlexVanderbist** [comment]
This was an issue on our end. Fix will be tagged in 2.2.0!
